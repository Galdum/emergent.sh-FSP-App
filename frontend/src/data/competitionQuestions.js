// Competition Questions and Configuration
// This file contains the quiz questions and competition settings for the leaderboard

export const competitionQuestions = {
  fachbegriffe: {
    title: "Fachbegriffe Flash",
    description: "Test rapid de termeni medicali germani",
    questions: [
      {
        question: "Ce înseamnă 'Anamnese' în germană?",
        options: [
          "Istoricul medical al pacientului",
          "Examenul fizic",
          "Diagnosticul",
          "Tratamentul"
        ],
        correct: 0
      },
      {
        question: "Care este traducerea corectă pentru 'Blutdruck'?",
        options: [
          "Tensiunea arterială",
          "Frecvența cardiacă",
          "Temperatura",
          "Glicemia"
        ],
        correct: 0
      },
      {
        question: "Ce înseamnă 'Herzinfarkt'?",
        options: [
          "Accident vascular cerebral",
          "Infarct miocardic",
          "Angină pectorală",
          "Aritmie"
        ],
        correct: 1
      },
      {
        question: "Care este termenul german pentru 'diabet'?",
        options: [
          "Diabetes",
          "Zuckerkrankheit",
          "Beide sunt corecte",
          "Niciunul"
        ],
        correct: 2
      },
      {
        question: "Ce înseamnă 'Schlaganfall'?",
        options: [
          "Infarct miocardic",
          "Accident vascular cerebral",
          "Epilepsie",
          "Migrenă"
        ],
        correct: 1
      }
    ]
  },
  diagnostic: {
    title: "Diagnostic Express",
    description: "Ghicește diagnosticul corect",
    questions: [
      {
        question: "Pacient cu durere în piept, dispnee și transpirație. Care este cel mai probabil diagnostic?",
        options: [
          "Infarct miocardic",
          "Pneumonie",
          "Gastrită",
          "Anxietate"
        ],
        correct: 0
      },
      {
        question: "Pacient cu cefalee intensă, fotofobie și rigiditate nucală. Care este diagnosticul?",
        options: [
          "Migrenă",
          "Meningită",
          "Hipertensiune",
          "Tumoră cerebrală"
        ],
        correct: 1
      },
      {
        question: "Pacient cu febră, tuse productivă și dispnee. Care este diagnosticul?",
        options: [
          "Bronșită acută",
          "Pneumonie",
          "Astm",
          "Tuberculoză"
        ],
        correct: 1
      }
    ]
  },
  grammar: {
    title: "Gramatică Germană",
    description: "Test rapid de gramatică germană",
    questions: [
      {
        question: "Care este forma corectă: 'Ich ___ Arzt'?",
        options: [
          "bin",
          "bist",
          "ist",
          "sind"
        ],
        correct: 0
      },
      {
        question: "Care este articolul corect pentru 'Krankenhaus'?",
        options: [
          "der",
          "die",
          "das",
          "den"
        ],
        correct: 2
      },
      {
        question: "Care este forma corectă: 'Ich gehe ___ Krankenhaus'?",
        options: [
          "zu",
          "in",
          "nach",
          "auf"
        ],
        correct: 1
      }
    ]
  }
};

export const competitionConfig = {
  competitions: [
    {
      id: 1,
      title: "Fachbegriffe Challenge",
      description: "Concurs rapid de termeni medicali germani",
      gameType: "fachbegriffe",
      status: "active",
      participants: 45,
      timeLimit: 30,
      reward: "50 XP",
      endsIn: "2h 15m",
      minScore: 4,
      bonusPoints: 25
    },
    {
      id: 2,
      title: "Diagnostic Master",
      description: "Testează-ți abilitățile de diagnostic",
      gameType: "diagnostic",
      status: "active",
      participants: 32,
      timeLimit: 45,
      reward: "75 XP",
      endsIn: "1h 30m",
      minScore: 2,
      bonusPoints: 35
    },
    {
      id: 3,
      title: "Grammar Sprint",
      description: "Quiz rapid de gramatică germană",
      gameType: "grammar",
      status: "starting_soon",
      participants: 28,
      timeLimit: 60,
      reward: "40 XP",
      startsIn: "30m",
      minScore: 2,
      bonusPoints: 20
    }
  ]
};