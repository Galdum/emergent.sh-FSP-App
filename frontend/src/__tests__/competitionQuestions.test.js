import { competitionQuestions, competitionConfig } from '../data/competitionQuestions';

describe('Competition Questions Data', () => {
  test('competitionQuestions should be defined and have required game types', () => {
    expect(competitionQuestions).toBeDefined();
    expect(competitionQuestions.fachbegriffe).toBeDefined();
    expect(competitionQuestions.diagnostic).toBeDefined();
    expect(competitionQuestions.grammar).toBeDefined();
  });

  test('each game type should have required properties', () => {
    const gameTypes = ['fachbegriffe', 'diagnostic', 'grammar'];
    
    gameTypes.forEach(gameType => {
      const game = competitionQuestions[gameType];
      expect(game.title).toBeDefined();
      expect(game.description).toBeDefined();
      expect(game.questions).toBeDefined();
      expect(Array.isArray(game.questions)).toBe(true);
      expect(game.questions.length).toBeGreaterThan(0);
    });
  });

  test('each question should have required properties', () => {
    Object.values(competitionQuestions).forEach(game => {
      game.questions.forEach(question => {
        expect(question.question).toBeDefined();
        expect(question.options).toBeDefined();
        expect(question.correct).toBeDefined();
        expect(Array.isArray(question.options)).toBe(true);
        expect(question.options.length).toBeGreaterThan(0);
        expect(typeof question.correct).toBe('number');
        expect(question.correct).toBeGreaterThanOrEqual(0);
        expect(question.correct).toBeLessThan(question.options.length);
      });
    });
  });

  test('competitionConfig should be defined and have competitions array', () => {
    expect(competitionConfig).toBeDefined();
    expect(competitionConfig.competitions).toBeDefined();
    expect(Array.isArray(competitionConfig.competitions)).toBe(true);
  });

  test('each competition should have required properties', () => {
    competitionConfig.competitions.forEach(competition => {
      expect(competition.id).toBeDefined();
      expect(competition.title).toBeDefined();
      expect(competition.description).toBeDefined();
      expect(competition.gameType).toBeDefined();
      expect(competition.status).toBeDefined();
      expect(competition.participants).toBeDefined();
      expect(competition.timeLimit).toBeDefined();
      expect(competition.reward).toBeDefined();
      expect(competition.minScore).toBeDefined();
      expect(competition.bonusPoints).toBeDefined();
    });
  });

  test('competition gameTypes should match available game types', () => {
    const availableGameTypes = Object.keys(competitionQuestions);
    
    competitionConfig.competitions.forEach(competition => {
      expect(availableGameTypes).toContain(competition.gameType);
    });
  });

  test('snapshot test for data structure', () => {
    expect(competitionQuestions).toMatchSnapshot();
    expect(competitionConfig).toMatchSnapshot();
  });
});