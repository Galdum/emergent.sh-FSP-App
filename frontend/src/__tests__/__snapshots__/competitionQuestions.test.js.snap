// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Competition Questions Data snapshot test for data structure 1`] = `
Object {
  "diagnostic": Object {
    "description": "Ghicește diagnosticul corect",
    "questions": Array [
      Object {
        "correct": 0,
        "options": Array [
          "Infarct miocardic",
          "Pneumonie",
          "Gastrită",
          "Anxietate",
        ],
        "question": "Pacient cu durere în piept, dispnee și transpirație. Care este cel mai probabil diagnostic?",
      },
      Object {
        "correct": 1,
        "options": Array [
          "Migrenă",
          "Meningită",
          "Hipertensiune",
          "Tumoră cerebrală",
        ],
        "question": "Pacient cu cefalee intensă, fotofobie și rigiditate nucală. Care este diagnosticul?",
      },
      Object {
        "correct": 1,
        "options": Array [
          "Bronșită acută",
          "Pneumonie",
          "Astm",
          "Tuberculoză",
        ],
        "question": "Pacient cu febră, tuse productivă și dispnee. Care este diagnosticul?",
      },
    ],
    "title": "Diagnostic Express",
  },
  "fachbegriffe": Object {
    "description": "Test rapid de termeni medicali germani",
    "questions": Array [
      Object {
        "correct": 0,
        "options": Array [
          "Istoricul medical al pacientului",
          "Examenul fizic",
          "Diagnosticul",
          "Tratamentul",
        ],
        "question": "Ce înseamnă 'Anamnese' în germană?",
      },
      Object {
        "correct": 0,
        "options": Array [
          "Tensiunea arterială",
          "Frecvența cardiacă",
          "Temperatura",
          "Glicemia",
        ],
        "question": "Care este traducerea corectă pentru 'Blutdruck'?",
      },
      Object {
        "correct": 1,
        "options": Array [
          "Accident vascular cerebral",
          "Infarct miocardic",
          "Angină pectorală",
          "Aritmie",
        ],
        "question": "Ce înseamnă 'Herzinfarkt'?",
      },
      Object {
        "correct": 2,
        "options": Array [
          "Diabetes",
          "Zuckerkrankheit",
          "Beide sunt corecte",
          "Niciunul",
        ],
        "question": "Care este termenul german pentru 'diabet'?",
      },
      Object {
        "correct": 1,
        "options": Array [
          "Infarct miocardic",
          "Accident vascular cerebral",
          "Epilepsie",
          "Migrenă",
        ],
        "question": "Ce înseamnă 'Schlaganfall'?",
      },
    ],
    "title": "Fachbegriffe Flash",
  },
  "grammar": Object {
    "description": "Test rapid de gramatică germană",
    "questions": Array [
      Object {
        "correct": 0,
        "options": Array [
          "bin",
          "bist",
          "ist",
          "sind",
        ],
        "question": "Care este forma corectă: 'Ich ___ Arzt'?",
      },
      Object {
        "correct": 2,
        "options": Array [
          "der",
          "die",
          "das",
          "den",
        ],
        "question": "Care este articolul corect pentru 'Krankenhaus'?",
      },
      Object {
        "correct": 1,
        "options": Array [
          "zu",
          "in",
          "nach",
          "auf",
        ],
        "question": "Care este forma corectă: 'Ich gehe ___ Krankenhaus'?",
      },
    ],
    "title": "Gramatică Germană",
  },
}
`;

exports[`Competition Questions Data snapshot test for data structure 2`] = `
Object {
  "competitions": Array [
    Object {
      "bonusPoints": 25,
      "description": "Concurs rapid de termeni medicali germani",
      "endsIn": "2h 15m",
      "gameType": "fachbegriffe",
      "id": 1,
      "minScore": 4,
      "participants": 45,
      "reward": "50 XP",
      "status": "active",
      "timeLimit": 30,
      "title": "Fachbegriffe Challenge",
    },
    Object {
      "bonusPoints": 35,
      "description": "Testează-ți abilitățile de diagnostic",
      "endsIn": "1h 30m",
      "gameType": "diagnostic",
      "id": 2,
      "minScore": 2,
      "participants": 32,
      "reward": "75 XP",
      "status": "active",
      "timeLimit": 45,
      "title": "Diagnostic Master",
    },
    Object {
      "bonusPoints": 20,
      "description": "Quiz rapid de gramatică germană",
      "gameType": "grammar",
      "id": 3,
      "minScore": 2,
      "participants": 28,
      "reward": "40 XP",
      "startsIn": "30m",
      "status": "starting_soon",
      "timeLimit": 60,
      "title": "Grammar Sprint",
    },
  ],
}
`;
